extension/src/openvic-extension/classes/GUINode.cpp

```diff
@@ -196,3 +233,125 @@ String GUINode::format_province_name(String const& province_identifier) {
 	static const String province_prefix = "PROV";
 	return province_prefix + province_identifier;
 }
+
+Ref<BitMap> GUINode::get_click_mask() const {
+	return _click_mask;
+}
+
+void GUINode::set_click_mask(Ref<BitMap> const& mask) {
+	if (_click_mask == mask) {
+		return;
+	}
+	_click_mask = mask;
+	queue_redraw();
+	update_minimum_size();
+}
+
+bool GUINode::_update_click_mask_for(Ref<Image> const& img, int index) {
+	ERR_FAIL_INDEX_V(index, _mask_controls.size(), false);
+	Control* control = _mask_controls[index];
+	if (!UtilityFunctions::is_instance_valid(control) && !control->is_inside_tree()) {
+		_mask_controls.remove_at(index);
+		return false;
+	}
+	ERR_FAIL_COND_V(img.is_null(), false);
+	Ref<Texture2D> texture = get_texture_from_node(control);
+	ERR_FAIL_COND_V(texture.is_null(), false);
+	Ref<Image> texture_img = texture->get_image();
+	if (img->is_empty()) {
+		img->copy_from(texture_img);
+	} else {
+		if (img->get_format() != texture_img->get_format()) {
+			img->convert(texture_img->get_format());
+		}
+		Vector2i img_size = img->get_size();
+		Vector2i total_size = control->get_screen_position() + texture_img->get_size();
+		Vector2i new_img_size = img_size.max(total_size);
+		if (new_img_size != img_size) {
+			img->crop(new_img_size.x, new_img_size.y);
+		}
+		img->blend_rect(texture_img, texture_img->get_used_rect(), control->get_position());
+	}
+	ERR_FAIL_COND_V(img->is_empty(), false);
+	return true;
+}
+
+void GUINode::update_click_mask() {
+	static constexpr real_t max_real = std::numeric_limits<real_t>::max();
+	static const Point2 max_point { max_real, max_real };
+	if (_mask_controls.size() == 0) {
+		return;
+	}
+
+	if (_click_mask.is_null()) {
+		_click_mask.instantiate();
+	}
+	Ref<Image> img;
+	img.instantiate();
+	Vector2 size = get_size();
+	img->create(size.x, size.y, false, Image::Format::FORMAT_RGBA8);
+	Point2 highest_position = { max_real, max_real };
+	for (int index = 0; index < _mask_controls.size(); index++) {
+		if (!_update_click_mask_for(img, index)) {
+			continue;
+		}
+		Vector2 screen_pos = _mask_controls[index]->get_screen_position();
+		if (screen_pos.x < highest_position.x && screen_pos.y < highest_position.y) {
+			highest_position = screen_pos;
+		}
```

Comment:
```suggestion
		highest_position = highest_position.min(screen_pos);
```
Using the min of each coordinate independently would better reflect how image size is calculated in `_update_click_mask_for`, and would make it order independent (e.g. for `(1, 2)` and `(2, 1)` whichever appears first would be chosen as the `highest_position` with the current condition).